test/storages/graph_storages/test_neo4j_graph.py

```diff
@@ -0,0 +1,333 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from unstructured.documents.elements import Element
+
+from camel.storages import Neo4jGraph
+from camel.storages.graph_storages.graph_element import (
+    GraphElement,
+    Node,
+    Relationship,
+)
+from camel.storages.graph_storages.neo4j_graph import (
+    BASE_ENTITY_LABEL,
+    NODE_PROPERTY_QUERY,
+    REL_PROPERTY_QUERY,
+    REL_QUERY,
+)
+
+test_data = [
+    GraphElement(
+        nodes=[Node(id="foo", type="foo"),
+               Node(id="bar", type="bar")], relationships=[
+                   Relationship(
+                       source=Node(id="foo", type="foo"),
+                       target=Node(id="bar", type="bar"),
+                       type="REL",
+                   )
+               ], source=Element(
+                   element_id="a04b820b51c760a41415c57c1eef8f08").to_dict())
+]
+
+NEO4J_URI = "neo4j+s://5af77aab.databases.neo4j.io"
+NEO4J_USERNAME = "neo4j"
+NEO4J_PASSWORD = "SEK_Fx5Bx-BkRwMx6__zM_TOPqXLWEP-czuIZ_u7-zE"
+
+
```

Comment:
Regarding there are more than 10 class methods needed to be tested, could you list the class methods tested by each function in the docstring? And it is good for the further dev.

like:

def test_cypher_return_correct_schema() -> None:
    r"""Test that chain returns direct results.
    Tested (what) class methods:
    *********
    """